Libraries/LibWeb/HTML/AutocompleteElement.cpp

```diff
@@ -0,0 +1,369 @@
+        //    then jump to the step labeled default.
+        if (index > maximum_tokens.value() - 1)
+            return step_default();
+
+        // 7. Set IDL value to the concatenation of the indexth token in tokens, a U+0020 SPACE character, and the previous value of IDL value.
+        idl_value = MUST(String::formatted("{} {}", tokens[index], idl_value));
+    }
+
+    // 17. If the indexth token in tokens is the first entry, then skip to the step labeled done.
+    if (index == 0)
+        return step_done();
+
+    // 18. Decrement index by one.
+    --index;
+
+    // 19. If category is Contact and the indexth token in tokens is an ASCII case-insensitive match for one of the strings in the following list,
+    //     then run the substeps that follow:
+    if (category == Category::Contact && tokens[index].to_ascii_lowercase().is_one_of("home", "work", "mobile", "fax", "pager")) {
+        // 1. Let contact be the matching string from the list above.
+        auto contact = tokens[index].to_ascii_lowercase();
+
+        // 2. Insert contact at the start of scope tokens.
+        scope_tokens.prepend(contact);
+
+        // 3. Add contact to hint tokens.
+        hint_tokens.append(contact);
+
+        // 4. Let IDL value be the concatenation of contact, a U+0020 SPACE character, and the previous value of IDL value.
+        idl_value = MUST(String::formatted("{} {}", contact, idl_value));
+
+        // 5. If the indexth entry in tokens is the first entry, then skip to the step labeled done.
+        if (index == 0)
+            return step_done();
+
+        // 6. Decrement index by one.
+        --index;
+    }
+
+    // 20. If the indexth token in tokens is an ASCII case-insensitive match for one of the strings in the following list,
+    //     then run the substeps that follow:
+    if (tokens[index].to_ascii_lowercase().is_one_of("shipping", "billing")) {
+        // 1. Let mode be the matching string from the list above.
+        auto mode = tokens[index].to_ascii_lowercase();
+
+        // 2. Insert mode at the start of scope tokens.
+        scope_tokens.prepend(mode);
+
+        // 3. Add mode to hint tokens.
+        hint_tokens.append(mode);
+
+        // 4. Let IDL value be the concatenation of mode, a U+0020 SPACE character, and the previous value of IDL value.
+        idl_value = MUST(String::formatted("{} {}", mode, idl_value));
+
+        // 5. If the indexth entry in tokens is the first entry, then skip to the step labeled done.
+        if (index == 0)
+            return step_done();
+
+        // 6. Decrement index by one.
+        --index;
+    }
+
+    // 21. If the indexth entry in tokens is not the first entry, then jump to the step labeled default.
+    if (index != 0)
+        return step_default();
+
+    // 22. If the first eight characters of the indexth token in tokens are not an ASCII case-insensitive match for the string "section-",
+    //     then jump to the step labeled default.
```

Comment:
do we not have a case insensitive "starts_with"?

Reply:
Not that I was able to find :thinking:

Reply:
FWIW, we have a `starts_with` function in `StringUtils` so you could do: `AK::StringUtils::starts_with(tokens[index], "section-"sv, CaseSensitivity::CaseInsensitive)`. 

I don't think there's much benefit though, since you do `to_ascii_lowercase()` of this value on the next line anyway.